lib/Conversion/LLVMFloatToFixed/Conversion.cpp

```diff
@@ -134,11 +136,18 @@ FloatToFixed::translateOrMatchOperand(Value *val, FixedPointType &iofixpt, Instr
   LLVM_DEBUG(dbgs() << "translateOrMatchOperand of " << *val << "\n");
   
   // FIXME: handle all the cases, we need more info about destination!
-  if (typepol == TypeMatchPolicy::ForceHint) {
-    LLVM_DEBUG(dbgs() << "translateOrMatchOperand: forcing hint as requested!\n");
+  if (typepol == TypeMatchPolicy::ForceHint || (!wasHintForced && hasInfo(val) && !fixPType(val).isSameTypeAs(iofixpt))) {
+    LLVM_DEBUG({
+      if (typepol == TypeMatchPolicy::ForceHint)
+        dbgs() << "translateOrMatchOperand: forcing hint as requested!\n";
+      else
+        dbgs() << "translateOrMatchOperand: mismatching conversion type, forcing conversion\n";
+    });
     FixedPointType origfixpt = iofixpt;
     llvm::Value *tmp = translateOrMatchOperand(val, iofixpt, ip, TypeMatchPolicy::RangeOverHintMaxFrac, true);
-    return genConvertFixedToFixed(tmp, iofixpt, origfixpt, ip);
+    llvm::Value *ret = genConvertFixedToFixed(tmp, iofixpt, origfixpt, ip);
+    iofixpt = origfixpt;
+    return ret;
```

Comment:
Ha senso questo pezzo?

In _mixed precision_, Quando c'è da tradurre una FDIV/FMUL con metadata fixed-point viene usato `TypeMatchPolicy::RangeOverHintMaxInt` anziché `TypeMatchPolicy::ForceHint`; ma se uno dei due operandi è float o posit deve essere riconvertito a fixed-point per poi fare l'operazione in fixed-point come richiesto, dunque ne forzo la conversione come se fosse stato passato `ForceHint`.

Inoltre al ritorno della funzione `iofixpt` deve contenere i metadati del dato ri-convertito, non quelli del valore trovato già precedentemente convertito in un tipo diverso.